crates/engine/tree/src/tree/root.rs

```diff
@@ -687,6 +687,11 @@ where
                     }
                     StateRootMessage::ProofCalculated(proof_calculated) => {
                         trace!(target: "engine::root", "processing StateRootMessage::ProofCalculated");
+                        let Some(sparse_trie_tx_ref) = sparse_trie_tx.as_ref() else {
+                            trace!(target: "engine::root", "ignoring proof calculation, tx dropped");
+                            continue
+                        };
```

Comment:
ah yeah this is because the exit condition of these two channels are a bit odd and we can drop the sender but keep looping

so this makes sense